packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -93,10 +93,24 @@ export class DockerImageAsset extends Construct implements assets.IAsset {
       exclude = [...exclude, ...fs.readFileSync(ignore).toString().split('\n').filter(e => !!e)];
     }
 
+    // include build context in "extra" so it will impact the hash
+    const extraHash = new Array<any>();
+    if (props.extra) { extraHash.push(props.extra); }
+    if (props.buildArgs) {
+      extraHash.push({ buildArgs: props.buildArgs });
+    }
+    if (props.target) {
+      extraHash.push({ target: props.target });
+    }
+    if (props.file) {
+      extraHash.push({ file: props.file });
+    }
+
```

Comment:
What about `props.repositoryName`? Though it doesn't affect the content of the image, it does affect its name. I think user's will find it surprising that changing the repo name will not have an affect on the asset.

Reply:
Added as well